File: proxy/tests/script_engine/url_protocol.lua
pcall(require, "../../script_libs/lua/roxy.lua")
---@type Extension
local url_scheme = {
	request = function(flow)
		if flow.request.url.protocol == Protocol.HTTP then
			flow.request.url.protocol = Protocol.HTTPS
		end
	end,
}
Extensions = { url_scheme }

File: proxy/tests/script_engine/url_protocol.py
from roxy import Extension, Protocol


class UrlScheme(Extension):
    def request(self, flow):
        print("protocol " + str(flow.request.url.protocol) + " " + str(Protocol.HTTP))
        if flow.request.url.protocol == Protocol.HTTP:
            print("setting https")
            flow.request.url.protocol = Protocol.HTTPS
        else:
            print("not eq")


Extensions = [UrlScheme()]
